content/posts/clean-your-backlog.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "Clean your backlog"
+date: 2021-11-10
+language: en
+author: Dimiter Petrov
+---
+
+Have you recently checked the size of your backlog?
+
+There is a limit to the amount of work a team can do at a given time, so new tasks get queued in a backlog. As a project grows, so does its backlog. There are new features to develop, more bugs to fix, but these tasks take longer to finish than to write down.
+
+## The problem with large backlogs
+
+Large backlogs are demoralizing. It is intimidating to realize how much is left to do.
+
+Large backlogs are overwhelming. People lose the overview. They create a second, smaller, "prioritized" backlog. Or they start tracking work through other channels: in-person meetings, email, text chat. That exacerbates the problem - now you have multiple backlogs and they all keep growing.
+
+Large backlogs stagnate. The bigger the backlog, the more information is outdated and the more maintenance is required.
+
+Large backlogs give a false sense of how much is really planned, because most items are too vague or not estimated yet.
+
+## Keep your backlog small(er)
+
+Reducing the backlog size can reduce the team's stress level. It also allows the team to focus on what's the most important. Here are some ideas to help you declutter:
+
+- **Acknowledge your backlog is too big.** They all are. Discuss that with your team and agree on what you want to do about it.
+- **Review the backlog as your priorities change.** What you though was important 4 months ago may not be as important now. Outright remove what has become irrelevant and reorder the rest to align with the new requirements.
+- **Drop items from the end of the backlog.** A little controversial, but if you keep the list in order of priority some items will always remain at the bottom. Why keep them at all if you never get around to them?
+- **Don't put technical debt tickets in the backlog.** Sometimes you identify code that would benefit from refactoring. If you can do it right away, just go ahead. If you don't have time, there is a good chance the project will have changed significantly by the time you do. The tech debt ticket will be outdated and you will need to assess the situation anew anyhow. On top of that, outdated tickets get further postponed if they are not actionable. My strategy is to address technical debt in small chunks as I work on a task affected by it, with the understanding that I will need more time.
+- **Separate your roadmap from your backlog.** Only bring the imminent and most concrete parts of the vision to the backlog.
```

Comment:
Maybe roadmap is the wrong term or the advice is too prescriptive. What do you think about the following phrasing?

> Don't plan too far ahead. Only bring the imminent and most concrete parts of the vision to the backlog. Consider keeping ideas which are not fully formed somewhere else.

It doesn't say where on purpose. I have used and heard about different solutions:
- a separate list as you suggest. I don't like if this is on the same project board (assuming a tool like Trello) because it also gets filled and most people don't look at it, but it takes space, also cognitive space.
- a separate board/project space. This could be an ideas board or a roadmap.
- a long-form text document in, say, Google Docs.

Reply:
> I agree that keeping the backlog small(er) is good. But I'm not sure if separating roadmap and backlog is a solution.

If you have a project that involves multiple components (e.g. front- and backend) it makes sense to have roadmap board where all epics reside and child tickets from each project are attached. This way you have a good overview what is going on across all things and depending on the project management tool, you can even assign resources and plan milestones without having to sync that information across different boards.

Reply:
@crackofdusk I like the phrasing and the argument for not saying where.

Reply:
Rephrased in ca7d693